### PR TITLE
feat!: add level generation pipeline

### PR DESCRIPTION
**Pull request by Tboefijn**
## Description

This PR puts the Level Generation in a pipeline class. This way it is easier to configure the level and is there less coupling between the classes.

## Testability

Open the scene `LevelGeneration`. Everything should still work as intended.

## Checklist
- [X] Set the proper pull request name
- [X] Merged main into your branch
- [X] Added a scene to the game for your changes

### CODE REVIEW

**Review by JensSteenmetz — CHANGES_REQUESTED** [review]
Aweeeesome work! I feel bad for the nitpick, but... skill issue

**Review by JensSteenmetz — DISMISSED** [review]
LGTMMMM

File: aplib.net-demo/Assets/Scenes/Settings.unity
Comment (JoenvandeVorle) [inline aplib.net-demo/Assets/Scenes/Settings.unity:1]: Pressing play in this scene doesn't seem to work for me

**Review by JoenvandeVorle — CHANGES_REQUESTED** [review]
nice! got a few comments

File: aplib.net-demo/Assets/Scenes/GridSystem.unity
[the remark this answers is not in the document]
Comment (Tboefijn) [reply aplib.net-demo/Assets/Scenes/GridSystem.unity:1]: It is outdated and I have deleted it now.

**Review by JensSteenmetz — APPROVED** [review]
LGTM!

**Review by JoenvandeVorle — APPROVED** [review]
lgtm, don't forget to delete your second branch as well

**Pull request merged**